Stats: Update copy on commercial purchase page

a8ck3n opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to p1HpG7-snd-p2.

## Proposed Changes

Updates the copy in the purchase flows depending on the site classification.

### Commercial sites see this

<img width="457" alt="SCR-20240523-qibx" src="https://github.com/Automattic/wp-calypso/assets/40267301/5ec22889-b9d9-42af-8843-f814f0bde4f1">

1. Updated header & description text
2. Updated styling for "commercial use" feature
3. New "Pick your tier" header
4. Updated copy in tier slider
5. Footnote indicator
6. Full-width buttons
7. Footnote text

### Non-commercial sites see this

<img width="456" alt="SCR-20240523-qjuq" src="https://github.com/Automattic/wp-calypso/assets/40267301/0f17480c-60aa-4bd9-994d-ac7aaa873619">

Same as above with different header, info, and button text.

### PWYW flow

<img width="438" alt="SCR-20240523-qgrm" src="https://github.com/Automattic/wp-calypso/assets/40267301/04a993cc-80d8-47c9-baf6-9598c8c98968">

1. Updated header & description text
2. Updated copy for UTM feature
8. Full-width buttons

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

Part of the project to improve the paywall experience.

p1HpG7-shw-p2

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Visit the Stats page on a new site and confirm the copy matches the second screenshot. New sites that aren't yet classified should see that.
* Visit the Stats page on a new site that has been classified as commercial (you can force this via the Blog RC tool) and confirm the copy matches the first screenshot.
* Confirm "Learn more" link is present and working when site is classified as commercial. Not pictured in screenshot.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

a8ck3n committed:
Introduce new copy based on site classification.

a8ck3n committed:
Move localization logic to helper function.

a8ck3n committed:
Comments++

a8ck3n committed:
Add “Learn more” link.

a8ck3n commented:
Added "Learn more" text with link to this public blog post:

https://jetpack.com/blog/updates-to-jetpack-stats-for-commercial-sites/

grzegorz-cp commented on client/my-sites/stats/stats-purchase/stats-purchase-single-item.tsx:
```diff
@@ -104,6 +104,46 @@ const StatsUpgradeInstructions = () => {
+
+	// Page title, info text, and button text depend on isCommercial status.
+	// We maintain the old behaviour for pre-FLAGS_CHECKOUT_FLOWS_V2 cases.
+	if ( ! config.isEnabled( FLAGS_CHECKOUT_FLOWS_V2 ) ) {
```
It looks like the feature flag has been released, so does it make sense to keep the condition?

grzegorz-cp commented on client/my-sites/stats/stats-purchase/stats-purchase-single-item.tsx:
```diff
@@ -104,6 +104,46 @@ const StatsUpgradeInstructions = () => {
+		continueButtonText = translate( 'Purchase' );
+	} else {
+		pageTitle = isCommercial
+			? translate( 'Upgrade and continue using Jetpack Stats' )
```
Instead of adding ternary operators everywhere, it would be easier to read the code if there was one `if` returning a set of the translations for commercial and another set for non-commercial visitors.

grzegorz-cp replied:
This will also avoid initialising the variables with an empty string.

grzegorz-cp commented on client/my-sites/stats/stats-purchase/stats-purchase-single-item.tsx:
```diff
@@ -104,6 +104,46 @@ const StatsUpgradeInstructions = () => {
+						comment: '{{link}} links to explainer post on Jetpack blog.',
+						components: {
+							link: (
+								<a href="https://jetpack.com/blog/updates-to-jetpack-stats-for-commercial-sites/" />
```
Let's put this in `const.js` in the main directory alongside other URLs.

a8ck3n committed:
Simplify logic for localized strings.

Removed the FLAGS_CHECKOUT_FLOWS_V2 code paths.

a8ck3n committed:
Remove FLAGS_CHECKOUT_FLOWS_V2 checks.

Flag has been released so we can clean up things a little bit.

a8ck3n committed:
Further simplify string localizations.

a8ck3n committed:
Replace raw URL with const.

a8ck3n commented:
Thanks for the feedback @grzegorz-cp. I've addressed those now. Going to review the feedback in Slack from design before requesting another review.

a8ck3n committed:
Updated copy for commercial flow.

a8ck3n committed:
Tweak styling in benefits listing.

Adds bold styling around the “Commercial use” list item.

a8ck3n committed:
Slider & button updates.

- Small slider copy updates
- Footnote regarding money-back guarantee
- Full width button

a8ck3n committed:
Update copy & layout for pwyw flow.

- Updated copy & button text
- Full-width buttons

a8ck3n committed:
Add simple header before tier slider.

Not a symantic header. Just a small bit of intro text to match the updated design.

grzegorz-cp commented on client/my-sites/stats/stats-purchase/stats-purchase-single-item.tsx:
```diff
@@ -166,6 +193,7 @@ const StatsCommercialPurchase = ( {
 			) }
 			{ isTierUpgradeSliderEnabled && (
 				<>
+					<p>Pick your Stats tier below:</p>
```
This needs to be translated.

grzegorz-cp commented on client/my-sites/stats/stats-purchase/stats-purchase-single-item.tsx:
```diff
@@ -190,6 +219,9 @@ const StatsCommercialPurchase = ( {
 						{ continueButtonText }
 					</ButtonComponent>
+					<div className="stats-purchase-flow__footnotes">
+						<p>(*) 14-day money-back guarantee</p>
```
This also needs to be translated.

grzegorz-cp commented on client/my-sites/stats/stats-purchase/stats-purchase-single-item.tsx:
```diff
@@ -104,6 +104,36 @@ const StatsUpgradeInstructions = () => {
 	);
 };
 
+function useLocalizedStrings( isCommercial: boolean ) {
```
The other functions in the file are arrow functions. Can you use the same type?

grzegorz-cp commented on client/my-sites/stats/stats-purchase/styles.scss:
```diff
@@ -589,3 +589,15 @@ $button-padding: 8px 32px;
 	margin-top: 24px;
 }
+
+.stats-purchase-flow__full-width-button {
```
Let's remove `button` from the name because it can be applied to other things.

grzegorz-cp commented on client/my-sites/stats/stats-purchase/stats-purchase-single-item.tsx:
```diff
@@ -174,6 +202,7 @@ const StatsCommercialPurchase = ( {
 						onSliderChange={ handleSliderChanged }
 					/>
 					<ButtonComponent
+						className="stats-purchase-flow__full-width-button"
```
`stats-purchase-flow` is a new prefix. Can you use the same one as the other elements?

grzegorz-cp reviewed (changes requested):
Looks good. Thanks for the updates. 
I left a few suggestions.

a8ck3n committed:
Wrap new UI strings in translate calls.

a8ck3n committed:
Update CSS classnames.

a8ck3n committed:
wibble

grzegorz-cp reviewed (approved):
Looks good 👍 
I hope that the last commit message for https://github.com/Automattic/wp-calypso/pull/90982/commits/19dbd319e0fdaeb4d08234bf75fa5285dd86ed52 doesn't describe the project.
